src/transaction/guards.ts

```diff
@@ -35,9 +35,12 @@ export const isTxInput = (
 export const isTxOutput = (
   test: any,
 ): test is TxOutput => {
-  if (!Array.isArray(test) || test.length !== 2) return false
-  const [address, coins] = test
-  return Buffer.isBuffer(address) && isLovelace(coins)
+  if (Array.isArray(test) && test.length === 2) {
+    const [address, amount] = test
+    const isMultiAsset = (coins: any) => Array.isArray(coins) && coins.length === 2
```

Comment:
Don't we want to also check if the given amounts are non-negative? And that policy ids have proper lengths and are buffers?

Reply:
As mentioned, the initial assumption was that all txs come from the `cardano-cli` so they should be valid. But I agree we should check for the buffer lengths I would make it into another PR tho.

Reply:
In general, it is probably OK to live with non-checked values in an initial release, but for the long term, the output of cardano-cli might change in the future, and if we don't validate thoroughly, we won't find out. (Or will find out in an opaque way, e.g. a generic Ledger error, which is hard to debug.)

Reply:
We will address this in another PR